### ISSUE TITLE
segfault writing to anonymous pipe

### ISSUE THREAD

**Issue by phemmer** [sahib/rmlint#212]
If you launch rmlint with the sh output going to an anonymous pipe, it segfaults.

For example:

    # rmlint -o sh:>(cat)
    ERROR: Aborting due to a fatal error. (signal received: Segmentation fault)
    ERROR: Please file a bug report (See rmlint -h)

Throwing an `strace` on shows it attempting to chase the anonymous pipe symlink:

```
open("/dev/fd/63", O_WRONLY|O_CREAT|O_TRUNC, 0666) = 3
lstat("/dev", {st_mode=S_IFDIR|0755, st_size=3220, ...}) = 0
lstat("/dev/fd", {st_mode=S_IFLNK|0777, st_size=13, ...}) = 0
readlink("/dev/fd", "/proc/self/fd", 4095) = 13
lstat("/proc", {st_mode=S_IFDIR|0555, st_size=0, ...}) = 0
lstat("/proc/self", {st_mode=S_IFLNK|0777, st_size=0, ...}) = 0
readlink("/proc/self", "5900", 4095)    = 4
lstat("/proc/5900", {st_mode=S_IFDIR|0555, st_size=0, ...}) = 0
lstat("/proc/5900/fd", {st_mode=S_IFDIR|0500, st_size=0, ...}) = 0
lstat("/proc/5900/fd/63", {st_mode=S_IFLNK|0300, st_size=64, ...}) = 0
readlink("/proc/5900/fd/63", "pipe:[55377455]", 4095) = 15
lstat("/proc/5900/fd/pipe:[55377455]", 0x7ffc53aa4ad0) = -1 ENOENT (No such file or directory)
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0} ---
```

Being able to do this would allow piping the output to bash and avoid the need for an intermediary file.

This is with version 2.2.0

**Comment by sahib** [comment]
Thank you for the report. I will look into a fix as soon as time permits.

**Comment by sahib** [comment]
Just a FYI: this will take a bit on my side, since I'm moving.

**Comment by sahib** [comment]
Should be fixed as of f3dae38. Still needs a testcase.

**Comment by sahib** [comment]
Closing, re-open when problems persists.
